src/tr2/game/savegame/savegame_bson.c

```diff
@@ -1193,33 +1191,31 @@ static bool M_LoadLara(JSON_OBJECT *const lara_obj)
     lara->hit_direction =
         JSON_ObjectGetInt(lara_obj, "hit_direction", lara->hit_direction);
     lara->air = JSON_ObjectGetInt(lara_obj, "air", lara->air);
-    lara->dive_count =
-        JSON_ObjectGetInt(lara_obj, "dive_count", lara->dive_count);
+    lara->dive_timer =
+        JSON_ObjectGetInt(lara_obj, "dive_count", lara->dive_timer);
     lara->death_timer =
         JSON_ObjectGetInt(lara_obj, "death_count", lara->death_timer);
     lara->current_active =
         JSON_ObjectGetInt(lara_obj, "current_active", lara->current_active);
     lara->hit_effect_count =
         JSON_ObjectGetInt(lara_obj, "hit_effect_count", lara->hit_effect_count);
-    lara->flare_age = JSON_ObjectGetInt(lara_obj, "flare_age", lara->flare_age);
-    lara->skidoo =
-        JSON_ObjectGetInt(lara_obj, "vehicle_item_number", lara->skidoo);
-    lara->back_gun = Object_UnmapGameID(
-        JSON_ObjectGetInt(lara_obj, "back_gun_obj_id", lara->back_gun));
-    lara->flare_frame =
-        JSON_ObjectGetInt(lara_obj, "flare_frame", lara->flare_frame);
+    lara->flare.age = JSON_ObjectGetInt(lara_obj, "flare_age", lara->flare.age);
+    lara->vehicle_item_num = JSON_ObjectGetInt(
+        lara_obj, "vehicle_item_number", lara->vehicle_item_num);
+    lara->back_gun_obj_id = Object_UnmapGameID(
+        JSON_ObjectGetInt(lara_obj, "back_gun_obj_id", lara->back_gun_obj_id));
+    lara->flare.frame_num =
+        JSON_ObjectGetInt(lara_obj, "flare_frame", lara->flare.frame_num);
     lara->mesh_effects =
         JSON_ObjectGetInt(lara_obj, "mesh_effects", lara->mesh_effects);
     lara->water_surface_dist = JSON_ObjectGetInt(
         lara_obj, "water_surface_dist", lara->water_surface_dist);
 
-    lara->flare_control_left = JSON_ObjectGetBool(
-        lara_obj, "flare_control_left", lara->flare_control_left);
-    lara->flare_control_right = JSON_ObjectGetBool(
-        lara_obj, "flare_control_right", lara->flare_control_right);
+    lara->flare.control =
+        JSON_ObjectGetBool(lara_obj, "flare_control", lara->flare.control);
```

Comment:
I noticed that we don't refer to the previous `flare_control_left` value. I don't mind removing it altogether if it has no consequences on the current saves, but if it creates problems, I'd add a fallback reading of the previous property with a note at which version we stopped supporting this.